packages/block-editor/src/store/selectors.js

```diff
@@ -1901,9 +1901,11 @@ const getAllAllowedPatterns = createSelector(
 	( state ) => {
 		const patterns = state.settings.__experimentalBlockPatterns;
 		const { allowedBlockTypes } = getSettings( state );
-		const parsedPatterns = patterns.map( ( { name } ) =>
-			__experimentalGetParsedPattern( state, name )
-		);
+		const parsedPatterns = patterns
+			.filter( ( { hideFromUI } ) => ! hideFromUI )
```

Comment:
I do wonder whether hideFormUI is the right name for it. Maybe at some point we'd want to show patterns in the UI somewhere but not in the inserter. Anyway, maybe we should just base it on how we do it for blocks. So (`inserter: false` or `supports.inserter: false`)

Reply:
I don't have strong opinion on this and I can change it. I just named it this way, as by using `inserter` made me think the `Inserter` (main+quick) where there might be more places that we do no want these patterns displayed like the [pattern explorer modal](https://github.com/WordPress/gutenberg/pull/35773) (if it lands 😄 ).